### ISSUE TITLE
demo for running OnACID in realtime

### ISSUE THREAD

**Issue by kushalkolar** [flatironinstitute/CaImAn#1043]
The OnACID demo shows how to use it for an existing file, but not for realtime data that are being streamed in, such as during image acquisition of an experiment.

I'm assuming that we should first call [`OnACID.initialize_online()`](https://caiman.readthedocs.io/en/master/core_functions.html#caiman.source_extraction.cnmf.online_cnmf.OnACID.initialize_online) followed by [`OnACID.fit_next()`](https://caiman.readthedocs.io/en/master/core_functions.html#caiman.source_extraction.cnmf.online_cnmf.OnACID.fit_next) for each frame as they come in? 

Myself and @clewis7 are working on an online acquisition, analysis & visualization system over the next several weeks and we could add a demo nb on how to do this. We're planning to use [shared memory buffers](https://docs.python.org/3/library/multiprocessing.shared_memory.html) for OnACID to fetch frames from, and dump motion corrected frames to for visualization.

**Comment by j-friedrich** [comment]
Of course, there is already an amzing demo for that :). The demo for running OnACID-E in real time is aptly named [demo_realtime_cnmfE.ipynb](https://github.com/flatironinstitute/CaImAn/blob/master/demos/notebooks/demo_realtime_cnmfE.ipynb).

The 2nd case illustrates what I think you want:  
We use a separate `Process` to acquire and add the images to a FIFO `Queue` at regular time intervals. The main process reads the next image from this `Queue` and waits for it if the `Queue` is empty.  
We first call [OnACID.initialize_online()](https://caiman.readthedocs.io/en/master/core_functions.html#caiman.source_extraction.cnmf.online_cnmf.OnACID.initialize_online) followed by [OnACID.mc_next()](https://github.com/flatironinstitute/CaImAn/blob/master/caiman/source_extraction/cnmf/online_cnmf.py#L1071) and  [OnACID.fit_next()](https://caiman.readthedocs.io/en/master/core_functions.html#caiman.source_extraction.cnmf.online_cnmf.OnACID.fit_next) for each frame as they come in.

To run OnACID instead of OnACID-E one would simply initialize using the low-rank background instead of the ring model.

**Comment by kushalkolar** [comment]
Thanks, this helps. There seem to be differences between OnACID for 2p and 1p, it's not easy to understand how to adapt the 1p example for 2p realtime use. I have some questions:

1. Is there a way to set an "infinite" number of `T` total frames, if the recording length is unknown? 

![image](https://user-images.githubusercontent.com/9403332/226098088-38398575-9d05-43e0-8c33-09de0c196530.png)

2. What are the different `init_method` options,  `"bare" | "cnmf" | "seeded"`. I notice `"bare"` is set for the 2p example, but `"cnmf"` is used for the realtime 1p example.

3. Are these two parameters relevant for 2p OnACID, I only see them in the 1p realtime example:

```
thresh_fitness_delta: float (negative)
Derivative test for detecting traces

thresh_fitness_raw: float (negative), default: computed from min_SNR
Threshold value for testing trace SNR
```

4. What is `model_LN` in `OnACID.initialize_online()`, can this be ignored if using `sniper_mode` with 2p data?

https://github.com/flatironinstitute/CaImAn/blob/8f61b060e90cab1888a8112259765f4b38775c47/caiman/source_extraction/cnmf/online_cnmf.py#L931-L938

Thanks! :smile:

**Comment by kushalkolar** [comment]
Another question, I'm assuming dendritic initialization doesn't exist so we can seed dendritic ourself and then use seeded?
